docs/contributor/03-40-kyma-bindings-processes.md

```diff
@@ -0,0 +1,75 @@
+3. KEB checks if the binding already exists. The binding in the database is identified by the Kyma instance ID and the binding ID, which are passed as a path query parameter. If the binding exists, KEB checks the values of the parameters of the existing binding. The OSB API requires a request to create a binding to fail if an object has already been created and the request contains different parameters.
+4. If the found binding is not expired, KEB returns it in the response. If the found binding is expired and exists in the database, KEB responds with an error and the Bad Request status. This check is done in an implicit database insert statement. The query fails for expired but existing bindings because the primary key is defined on the instance and binding IDs, not the expiration date. This is the case until the cleanup job removes the expired binding from the database. If the binding does not exist, the flow returns to the process's execution path, where no bindings exist in the database.
+5. Whether the binding exists or not, the last step in the request validation is to verify the number of bindings. Every instance is allowed to create a limited number of active bindings. The limit is configurable and, by default, set to 10 non-expired bindings. If the limit is not exceeded and the binding does not exist in the database, KEB proceeds to the next phase of the process - binding creation.
+
+## Binding Creation
+
+Binding creation part consists of the following steps:
+1. KEB creates a service binding object and generates a kubeconfig file with a JWT token. The kubeconfig file is valid for a specified period, defaulted or set in the request body.
+
+> [!NOTE]
+>  Expired bindings do not count towards the bindings limit. However, as long as they exist in the database, they prevent creating new bindings with the same ID. Only after they are removed by the cleanup job or manually can the binding be recreated again.
+
+2. KEB creates ServiceAccount, ClusterRole (administrator privileges), and ClusterRoleBinding, all named `kyma-binding-{{binding_id}}`. You can use the ClusterRole to modify permissions granted to the kubeconfig.
+3. The created resources are used to generate a [TokenRequest](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/). The token is wrapped in a kubeconfig template and returned to the user.
+4. The encrypted credentials are stored as an attribute in the previously created database binding.
+
+> [!NOTE]
+>  It is not recommended to create multiple and unused TokenRequest resources.
+
+
+# The Process of Fetching a Kyma Binding
+
+![Get Binding Flow](../assets/bindings-get-flow.drawio.svg)
+
+The process starts with a GET request sent to the KEB API.
+The first instruction in the process is to check if a Kyma instance exists. If a Kyma instance exists, it must not be deprovisioned or suspended.
```

Comment:
```suggestion
The first instruction in the process is to check if a Kyma instance exists. The found Kyma instance must not be deprovisioned or suspended.
```

Reply:
Who/what gives the instruction? Can we make this active?

Reply:
Changed to active voice.